chat.go

```diff
@@ -85,6 +85,17 @@ const (
 	FinishReasonNull          FinishReason = "null"
 )
 
+func (r FinishReason) MarshalJSON() ([]byte, error) {
```

Comment:
@yuikns Thanks for the suggestion. Sorry for the late reply.
The premise is that `FinishReason` is used only in responses. Therefore, there is usually no need to marshal it. Could you tell us in what situations you would like to marshal the response?

Reply:
@vvatanabe 
thank you for your attention.
in some of our projects, it is playing the role of gateway. we need to make sure it will correctly forward the message in the same way.
this string FinishReason broken the assumption.

Reply:
@yuikns I see. Understood. Thank you.
The cause of this issue is that the nullable value finish_reason doesn't allow null. Changing the field that uses FinishReason to a pointer might resolve this.

However, turning a primitive string value into a pointer might degrade the developer experience.

Your approach might be a good compromise.

On a different note, currently when finish_reason is null, it gets unmarshalled to an empty string. So, there's no chance that the "null" string indicated by FinishReasonNull is assigned to the FinishReason field, right?

Reply:
@vvatanabe 
exactly, I do not want to alter the API, particularly as changing a value type may cause some breaking change, which is hurting. This implementation could meets my requirements, allowing me to avoid any ugly workarounds.

Thank you for your understanding.